data-mongodb/src/main/java/io/micronaut/data/mongodb/database/MongoReactiveFactory.java

```diff
@@ -0,0 +1,119 @@
+        return mongoConfiguration.getConnectionString()
+                .map(ConnectionString::getDatabase)
+                .filter(StringUtils::isNotEmpty)
+                .map(databaseName -> new SimpleReactiveMongoDatabaseFactory(mongoClient, databaseName))
+                .orElseThrow(() -> new ConfigurationException("Please specify the default Mongo database in the url string"));
+    }
+
+    @EachBean(NamedMongoConfiguration.class)
+    @Singleton
+    ReactiveMongoDatabaseFactory namedMongoDatabaseFactory(NamedMongoConfiguration mongoConfiguration, @Parameter MongoClient mongoClient) {
+        return mongoConfiguration.getConnectionString()
+                .map(ConnectionString::getDatabase)
+                .filter(StringUtils::isNotEmpty)
+                .map(databaseName -> new SimpleReactiveMongoDatabaseFactory(mongoClient, databaseName))
+                .orElseThrow(() -> new ConfigurationException("Please specify the default Mongo database in the url string"));
+    }
+
+    @EachBean(DefaultReactiveMongoRepositoryOperations.class)
+    @Singleton
+    MongoRepositoryOperations syncOperations(DefaultReactiveMongoRepositoryOperations reactiveOperations) {
+        return new MongoReactiveBlockingRepositoryOperations(reactiveOperations);
+    }
+
+    private static final class MongoReactiveBlockingRepositoryOperations implements
+            MongoRepositoryOperations, BlockingReactorRepositoryOperations, AsyncCapableRepository {
+
+        private final DefaultReactiveMongoRepositoryOperations reactiveOperations;
+        private ExecutorService executorService;
+        private ExecutorAsyncOperations asyncOperations;
+
+        private MongoReactiveBlockingRepositoryOperations(DefaultReactiveMongoRepositoryOperations reactiveOperations) {
+            this.reactiveOperations = reactiveOperations;
+        }
+
+        @Override
+        public ReactorReactiveRepositoryOperations reactive() {
+            return reactiveOperations;
+        }
+
+        @Override
+        public ApplicationContext getApplicationContext() {
+            return reactiveOperations.getApplicationContext();
+        }
+
+        @NonNull
+        @Override
+        public ExecutorAsyncOperations async() {
+            ExecutorAsyncOperations asyncOperations = this.asyncOperations;
+            if (asyncOperations == null) {
+                synchronized (this) { // double check
+                    asyncOperations = this.asyncOperations;
+                    if (asyncOperations == null) {
+                        if (executorService == null) {
+                            executorService = Executors.newCachedThreadPool();
+                        }
+                        asyncOperations = new ExecutorAsyncOperations(this, executorService);
+                        this.asyncOperations = asyncOperations;
+                    }
+                }
+            }
+            return asyncOperations;
```

Comment:
Shouldn't the I/O thread pool be used for this?

Reply:
We use `newCachedThreadPool` in 4 other implementations of async operations.